code/modules/battlemonsters/items/deck.dm

```diff
@@ -0,0 +1,157 @@
+/obj/item/battle_monsters/deck
+	name = "battle monsters deck"
+	icon_state = "stack"
+	w_class = ITEMSIZE_SMALL
+	var/list/stored_card_names = list()
+	var/deck_size = 52
+
+/obj/item/battle_monsters/deck/proc/get_top_card()
+	return stored_card_names[stored_card_names.len]
+
+/obj/item/battle_monsters/deck/proc/get_bottom_card()
+	return stored_card_names[1]
+
+/obj/item/battle_monsters/deck/proc/add_card(var/user, var/obj/item/battle_monsters/card/added_card)
+
+	if(added_card.spell_datum)
+		stored_card_names += "spell_type,[added_card.spell_datum.id],no_title"
+	else if(added_card.trap_datum)
+		stored_card_names += "trap_type,[added_card.trap_datum.id],no_title"
+	else
+		stored_card_names += "[added_card.prefix_datum.id],[added_card.root_datum.id],[added_card.suffix_datum.id]"
+
+	qdel(added_card)
+
+/obj/item/battle_monsters/deck/proc/take_card(var/mob/user)
+
+	if(icon_state == "stack") //Deck form
+		user.visible_message(\
+			span("notice","\The [user] draws a card from their [src]."),\
+			span("notice","You draw a card from their [src].")\
+		)
+	else
+		user.visible_message(\
+			span("notice","\The [user] takes a card from their hand."),\
+			span("notice","You take a card from your hand.")\
+		)
+
+	take_specific_card(user,get_top_card())
+
+/obj/item/battle_monsters/deck/proc/take_specific_card(var/mob/user, var/card_id)
+	if(card_id in stored_card_names)
+		var/obj/item/battle_monsters/card/new_card = SSbattlemonsters.CreateCard(card_id,src.loc)
+
+		if(!user.get_active_hand())
+			user.put_in_active_hand(new_card)
+		else if(!user.get_inactive_hand())
+			user.put_in_inactive_hand(new_card)
+		else
+			to_chat(user,span("notice","Your hands are full!"))
+			return
+
+		new_card.pickup(user)
+		stored_card_names -= card_id
+
+	if(stored_card_names.len <= 0)
+		qdel(src)
+
+/obj/item/battle_monsters/deck/MouseDrop_T(var/atom/movable/C, mob/user) //Dropping C onto the card
+	if(istype(C,/obj/item/battle_monsters/deck/) && C.loc == loc)
+		var/obj/item/battle_monsters/deck/added_deck = C
+		added_deck.stored_card_names += stored_card_names
+
+		user.visible_message(\
+			span("notice","\The [user] combines two decks together."),\
+			span("notice","You combine too decks together.")\
+		)
+
+		qdel(src)
+		return
+
+	. = ..()
+
+/obj/item/battle_monsters/deck/verb/toggle_mode()
```

Comment:
Might be a good idea to throw a use_check() into this verb.

Reply:
Done